bin/check-dns.rb

```diff
@@ -96,15 +96,39 @@ class DNS < Sensu::Plugin::Check::CLI
          long: '--use-tcp',
          boolean: true
 
+  option :count,
+         description: 'Number of times to try resolution',
+         short: '-c COUNT',
+         long: '--count COUNT',
+         proc: proc(&:to_i),
+         default: 1
+
+  option :threshold,
+         description: 'Percentage of DNS queries that must succeed',
+         short: '-l PERCENT',
+         long: '--threshold PERCENT',
+         proc: proc(&:to_i),
+         default: 100
+
   def resolve_domain
     dnsruby_config = {}
     dnsruby_config[:nameserver] = [config[:server]] unless config[:server].nil?
     dnsruby_config[:port] = config[:port] unless config[:port].nil?
     dnsruby_config[:use_tcp] = config[:use_tcp] unless config[:use_tcp].nil?
     resolv = Dnsruby::Resolver.new(dnsruby_config)
     resolv.do_validation = true if config[:validate]
-    entries = resolv.query(config[:domain], config[:type], config[:class])
-    puts "Entries: #{entries}" if config[:debug]
+    entries = []
+    count = 0
+    while count < config[:count]
```

Comment:
we may want to validate that they pass in whole numbers greater than 0. This would not work if someone accidentally passed in 0 or a negative number.